### PR TITLE
POS Update

### PR DESCRIPTION
**Pull request by 4855062503**

### CODE REVIEW

File: test_pos/app.html [hunk 1]
```diff
@@ -2,67 +2,209 @@
 <html lang="en-US">
     <head>
         <meta charset="utf-8">
+      <style>
```
Comment (djpnewton) [inline test_pos/app.html]: I think it would be nicer to move this into a seperate css file

File: test_pos/app.html [hunk 2]
```diff
@@ -2,67 +2,209 @@
...
+	  const WS_URL = "https://premio-demo.caprover.acuerdo.dev/paydb"
 
-<script src="https://cdnjs.cloudflare.com/ajax/libs/crypto-js/3.1.2/rollups/hmac-sha256.js" integrity="sha384-nIoggK09VaqN3XWoPnmd+/rQnyaDFUUqYYH1e3S0h7w5Q/q8eUBbNUTaI2auJiQz" crossorigin="anonymous"></script>
-<script src="https://cdnjs.cloudflare.com/ajax/libs/crypto-js/3.1.2/components/enc-base64-min.js" integrity="sha384-QhR/8T3ugj5vlN+fMUechKVmirQsSCuvX+EZRHVe3SCVhyQyy/VxxlP2Qme8iFCr" crossorigin="anonymous"></script>
-<script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.1.2/socket.io.min.js" integrity="sha384-toS6mmwu70G0fw54EGlWWeA4z3dyJ+dlXBtSURSKN4vyRFOcxd3Bzjj/AoOwY+Rg" crossorigin="anonymous"></script>
-<script>
-    const URL_BASE = "https://premio-demo.caprover.acuerdo.dev/"
-    const WS_URL = "https://premio-demo.caprover.acuerdo.dev/paydb"
-
-    const apikey = 'XXXXX';
```
Comment (djpnewton) [inline test_pos/app.html:14]: read from localStorage, also have a button to update what is in localStorage

File: test_pos/app.html [hunk 3]
```diff
@@ -2,67 +2,209 @@
...
+          const REBATE_FACTOR = 0.05;
+          var qrCodeObj = null;
+          var email;
+          var photo;
+          var photoType;
+
+          function updateQr() {
+              var length = window.innerWidth * 0.25;
+              var logoSrc = "data:image/png;base64," + photo;
+              //TODO if photoType =- 'svg'
```
Comment (djpnewton) [inline test_pos/app.html]: handle the case of SVG

File: test_pos/app.html [hunk 4]
```diff
@@ -2,67 +2,209 @@
...
+                        }
+                      ).then(
+                          function(emailInput) {
+                            postPayDb('payment_create', {reason: "rebate", recipient: emailInput.value, amount: Math.floor(tx.amount * REBATE_FACTOR), category: "testing", message: 1})
+                            .then(
+                              function(res) {
+                                console.log("sent");
+                              }
+                            )
+
```
Comment (djpnewton) [inline test_pos/app.html]: fix up weird spacing here

File: test_pos/app.html [hunk 5]
```diff
@@ -2,67 +2,209 @@
...
+                            }
+                            return rebateEmail; 
+                          }
+                        }
+                      ).then(
+                          function(emailInput) {
+                            postPayDb('payment_create', {reason: "rebate", recipient: emailInput.value, amount: Math.floor(tx.amount * REBATE_FACTOR), category: "testing", message: 1})
+                            .then(
+                              function(res) {
+                                console.log("sent");
```
Comment (djpnewton) [inline test_pos/app.html]: you should validate the amount and invoice id, also show a nice popup to show the rebate succeeded

File: test_pos/app.html [hunk 6]
```diff
@@ -2,67 +2,209 @@
...
+                  function(result) {
+                    if(result.isConfirmed) {
+                      Swal.fire(
+                        {
+                          title: "give rebate",
+                          html: `<input type='email' id='email-input' class='swal2-input' placeholder='email'>`,
+                          preConfirm: function() {
+                            const rebateEmail = Swal.getPopup().querySelector("#email-input").value;
+                            if(!rebateEmail) {
+                              Swal.showValidationMessage('please enter an email');
```
Comment (djpnewton) [inline test_pos/app.html]: have an option to scan qr code

**Review by djpnewton — CHANGES_REQUESTED** [review]

**Comment by djpnewton** [comment]
you should be developing from a feature branch, otherwise you will get fucked up when other commits get pushed to master

**Comment by djpnewton** [comment]
the spacing of your JS is not consistent

**Comment by djpnewton** [comment]
> the spacing of your JS is not consistent

@4855062503 can you address this

**Comment by 4855062503** [comment]
> > the spacing of your JS is not consistent
> 
> @4855062503 can you address this

Sure.

**Pull request closed without merging**